ern-core/src/CodePushSdk.js

```diff
@@ -22,13 +22,20 @@ export type CodePushReleaseInfo = {
   /* generated */ uploadTime: number;
 };
 
+export type CodePushInitConfig = {
+  accessKey: string,
+  customHeaders?: {[headerName: string]: string},
+  customServerUrl?: string,
+  proxy?: string
+}
+
 export type CodePushPackage = CodePushPackageInfo & CodePushReleaseInfo;
 
 export default class CodePushSdk {
   _codePush: any
 
-  constructor (accessKey: string) {
-    this._codePush = new CodePush(accessKey)
+  constructor (initConfig: CodePushInitConfig) {
+    this._codePush = new CodePush(initConfig.accessKey, initConfig.customHeaders, initConfig.customServerUrl, initConfig.proxy)
```

Comment:
is it possible that properties may be undefined/null? In that case , let MSFT code-push client take care of default value and restrict the properties sent.

Reply:
@krunalsshah 
this will happen anyway for any `undefined` as if let's say only access key is specified, the call will end up being `new CodePush('ACCESS_KEY', undefined, undefined, undefined)` which JS will translate anyway to `new CodePush('ACCESS_KEY')`. 
I would just add a check if `accessKey` is not undefined/null or empty string and throw here, as it is the only mandatory parameter.

Reply:
Access is validated here https://github.com/electrode-io/electrode-native/pull/683/files#diff-659728c901f8b033afe49b670ea0229aR568